sql/python/models/example.py

```diff
@@ -0,0 +1,93 @@
+
+train, test = train_test_split(dataframe, test_size=0.2)
+train, val = train_test_split(train, test_size=0.2)
+print(len(train), 'train examples')
+print(len(val), 'validation examples')
+print(len(test), 'test examples')
+
+# A utility method to create a tf.data dataset from a Pandas Dataframe
+def df_to_dataset(dataframe, shuffle=True, batch_size=1):
+    dataframe = dataframe.copy()
+    labels = dataframe.pop('target')
+    ds = tf.data.Dataset.from_tensor_slices((dict(dataframe), labels))
+    if shuffle:
+        ds = ds.shuffle(buffer_size=len(dataframe))
+    ds = ds.batch(batch_size)
+    return ds
+
+batch_size = 32 # A small batch sized is used for demonstration purposes
+train_ds = df_to_dataset(train, batch_size=batch_size)
+val_ds = df_to_dataset(val, shuffle=False, batch_size=batch_size)
+test_ds = df_to_dataset(test, shuffle=False, batch_size=batch_size)
+
+feature_columns = [tf.feature_column.numeric_column(header) for header in ['c1', 'c2', 'c3', 'c4', 'c5']]
+
+class DNNClassifier(tf.keras.Model):
+    def __init__(self, feature_columns, hidden_units, n_classes):
+        """DNNClassifier
+        :param feature_columns: feature columns.
+        :type feature_columns: list[tf.feature_column].
+        :param hidden_units: number of hidden units.
+        :type hidden_units: list[int].
+        :param n_classes: List of hidden units per layer.
+        :type n_classes: int.
+        """
+        super(DNNClassifier, self).__init__()
+
+        # combines all the data as a dense tensor
+        self.feature_layer = tf.keras.layers.DenseFeatures(feature_columns)
+        self.hidden_layers = []
+        for hidden_unit in hidden_units:
+            self.hidden_layers.append(tf.keras.layers.Dense(hidden_unit))
+        self.prediction_layer = tf.keras.layers.Dense(n_classes, activation='softmax')
+
+    def call(self, inputs):
+        x = self.feature_layer(inputs)
+        for hidden_layer in self.hidden_layers:
+            x = hidden_layer(x)
+        return self.prediction_layer(x)
+
+    def default_optimizer(self):
+        """Default optimizer name. Used in model.compile."""
+        return 'adam'
+
+    def default_loss(self):
+        """Default loss function. Used in model.compile."""
+        return 'categorical_crossentropy'
+
+    def default_training_epochs(self):
+        """Default training epochs. Used in model.fit."""
+        return 5
+
+    def prepare_prediction_column(self, prediction):
+        """Return the class label of highest probability."""
+        return prediction.argmax(axis=-1)
+
+model = DNNClassifier(feature_columns=feature_columns, hidden_units=[10, 10], n_classes=3)
+model.compile(optimizer=model.default_optimizer(), loss=model.default_loss())
+
+is_training = False
+if is_training:
+    model.fit(train_ds, validation_data=val_ds, epochs=model.default_training_epochs(), verbose=0)
+    model.save_weights('my_model.h5')
+    print("Done training.")
+else:
+    model.predict(test_ds)
```

Comment:
Because DenseFeature is a new feature in tf 2.0alpha0, it is not mature. model.build(input_shape) cannot accept dict of shapes, which is required by DenseFeature.
I believe when 2.0 releases, it should support:
```
    shape = {'c1': (1,1), 'c2': (1,1), 'c3': (1,1), 'c4': (1,1), 'c5': (1,1)}
    model.build(shape)
```

I modified tf code `tensorflow/python/keras/engine/network.py` to add dict support, in `def build(self, input_shape):`
```
    # hack to add dict
    valid_types = (tuple, list, dict, tensor_shape.TensorShape)
    if not isinstance(input_shape, valid_types):
      raise ValueError('Specified input shape is not one of the valid types. '
                       'Please specify a batch input shape of type tuple or '
                       'list of input shapes. User provided '
                       'input type: {}'.format(type(input_shape)))

    if input_shape and not self.inputs:
      # We create placeholders for the `None`s in the shape and build the model
      # in a Graph. Since tf.Variable is compatible with both eager execution
      # and graph building, the variables created after building the model in
      # a Graph are still valid when executing eagerly.
      if context.executing_eagerly():
        graph = func_graph.FuncGraph('build_graph')
      else:
        graph = backend.get_graph()
      with graph.as_default():
        # hack
        if isinstance(input_shape, dict):
            x = {key: base_layer_utils.generate_placeholders_from_shape(input_shape[key])
               for key in input_shape}
        elif isinstance(input_shape, list):
          x = [base_layer_utils.generate_placeholders_from_shape(shape)
               for shape in input_shape]
        else:
          x = base_layer_utils.generate_placeholders_from_shape(input_shape)
```
After that, using dict of shapes as input for model.build works.

Reply:
@skydoorkai This workaround looks great. Is it possible to contribute this feature to Tensorflow official repo?

Reply:
We can file an issue for this. Not sure if the workaround is the right solution for it.

Reply:
@skydoorkai could you share your modification? It is good to have it in your personal repo or in an issue in the TensorFlow repo. Anyway, others could understand the requirement and your solution.

Reply:
I will create a dockerfile in my repo and file an issue in tensorflow repo.

Reply:
Put the repro example and the solution here:
https://github.com/skydoorkai/tests_and_issues/tree/master/tensorflow/densefeature

Also filed a tf bug:
https://github.com/tensorflow/tensorflow/issues/28111

Reply:
This workaround looks good to me. Do you want to make this a PR to TF? Or do you want us to add it?